x/authenticator/README.md

```diff
@@ -121,3 +132,40 @@ The anyOf authenticator allows you to specify a list of authenticators. If any o
 
 The allOf authenticator allows you to specify a list of authenticators. All authenticators in the list must successfully authenticate a message for the message to be authenticated.
 
+## CosmWasm Authenticator
+
+The CosmWasm Authenticator allows for the building of any custom authentication logic as a CosmWasm contract.
+The contract needs to be instantiated before being added as an authenticator, since the contract address is required for `MsgAddAuthenticator`.
+
+`MsgAddAuthenticator` arguments:
+
+```
+sender: <bech32_address>
+type: "CosmwasmAuthenticatorV1"
+data: json_bytes({
+    contract: "<contract_address>",
+    params: [<byte_array>]
+})
+```
+
+The `params` field is for configuring any additional parameters for the authenticator and is _encouraged over using contract storage_ when possible
+since the field is standardized and must be json bytes. It can be easily parsed and displayed by clients, which makes them more human-friendly.
```

Comment:
```suggestion
The `params` field is for configuring any additional parameters for the authenticator and can be used to pass user specific information to the contract instead of keeping it in contract storage. This allows for contract reuse even even when user specific information is needed. 
This field is standardized and must be json bytes. This way it can be easily parsed and displayed by clients, which makes them more human-friendly.
```